backend/sites/src/app/services/site/site.service.ts

```diff
@@ -1116,4 +1129,540 @@ export class SiteService {
+
+        if (searchParam.addrLine != null && searchParam.addrLine != '') {
+          result = result.filter((res) => {
+            return res.address.indexOf(searchParam.addrLine) !== -1;
+          });
+        }
+
+        if (searchParam.whoCreated != null && searchParam.whoCreated != '') {
+          result = result.filter((res) => {
+            return (
+              res.whoUpdated
+                .toLowerCase()
+                .indexOf(searchParam.whoCreated.toLowerCase()) !== -1
+            );
+          });
+        }
+      }
+
+      const startIndex = (page - 1) * pageSize;
+
+      const paginatedRecords = result.slice(startIndex, startIndex + pageSize);
+
+      const resultDTO: QueryResultForPendingSites = {
+        data: paginatedRecords,
+        totalRecords: result.length,
+      };
+
+      this.sitesLogger.log('SiteService.getSiteDetailsPendingSRApproval() end');
+      this.sitesLogger.debug(
+        'SiteService.getSiteDetailsPendingSRApproval() end',
+      );
+
+      return resultDTO;
+    } catch (error) {
+      this.sitesLogger.log(
+        'SiteService.getSiteDetailsPendingSRApproval() error' +
+          JSON.stringify(error),
+      );
+
+      throw new Error('Failed to determine banner type.');
+    }
+  }
+
+  /**
+   * Bulk Approval/ Reject For Site Registry Approvals
+   * @param inputDTO list of site
+   * @param userInfo authenticated user
+   * @returns true/false
+   */
+  async bulkUpdateForSR(
+    inputDTO: BulkApproveRejectChangesDTO,
+    userInfo: any,
+  ): Promise<boolean> {
+    try {
+      this.sitesLogger.log('SiteService.bulkUpdateForSR() start');
+      this.sitesLogger.debug('SiteService.bulkUpdateForSR() start');
+
+      if (!inputDTO) {
+        return false;
+      } else {
+        const { isApproved, sites } = inputDTO;
+
+        sites.forEach(async (site: SiteRecordsForSRAction) => {
+        await this.entityManager.transaction(
+          async (transactionalEntityManager: EntityManager) => {            
+              if (
+                (site != null && site.siteId === null) ||
+                site.siteId === undefined
+              ) {
+                return false;
+              }
+              await this.processSRBulkUpdates(
+                transactionalEntityManager,
+                site,
+                isApproved,
+                userInfo,
+              );
+            });
+          },
+        );
+
+        return true;
+      }
+    } catch (error) {
+      this.sitesLogger.log(
+        'SiteService.bulkUpdateForSR() error' + JSON.stringify(error),
+      );
+      return false;
+    }
+  }
+
+  async processSRBulkUpdates(
+    transactionalEntityManager: EntityManager,
+    site: SiteRecordsForSRAction,
+    isApproved: boolean,
+    userInfo: any,
+  ) {
+    try {
+      this.sitesLogger.log('SiteService.processSRBulkUpdates() start');
+
+      if ((site != null && site.siteId === null) || site.siteId === undefined) {
+        return false;
```

Comment:
if this is an exit point for the method, pls  have the logger log the end of the method

Reply:
done